bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -966,7 +970,17 @@ public void run() {
                             forceWriteBatchBytesStats.registerSuccessfulValue(batchSize);
 
                             boolean shouldRolloverJournal = (lastFlushPosition > maxJournalSize);
-                            forceWriteRequests.put(createForceWriteRequest(logFile, logId, lastFlushPosition, toFlush, shouldRolloverJournal, false));
+                            if (syncData) {
+                                // Trigger data sync to disk in the "Force-Write" thread. Callback will be triggered after data is committed to disk
+                                forceWriteRequests.put(createForceWriteRequest(logFile, logId, lastFlushPosition, toFlush, shouldRolloverJournal, false));
+                            } else {
+                                // Data is already written on the file (though it might still be in the OS page-cache)
```

Comment:
When the flag is false, are we completely letting it to the os to flush it? Should we have a background thread to flush at some interval?

Reply:
For now, I'd like to try out with this option. Since it's not guaranteeing the data durability in case of power failure anyway, we could see later whether we want to reduce the time window.